pkg/controller/kogitoinfra/mongodb.go

```diff
@@ -0,0 +1,330 @@
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      mongoDBSecretName,
+			Namespace: namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		StringData: map[string]string{
+			infrastructure.MongoDBAppSecretAuthDatabaseKey: credentials.AuthDatabase,
+			infrastructure.MongoDBAppSecretUsernameKey:     credentials.Username,
+			infrastructure.MongoDBAppSecretPasswordKey:     credentials.Password,
+			infrastructure.MongoDBAppSecretDatabaseKey:     credentials.Database,
+		},
+	}
+	if err := framework.SetOwner(i.instance, i.scheme, secret); err != nil {
+		return nil, err
+	}
+	if err := kubernetes.ResourceC(i.client).Create(secret); err != nil {
+		log.Errorf("Error occurs while creating %s", secret)
+		return nil, err
+	}
+	log.Debug("%s successfully created", secret)
+	return secret, nil
+}
+
+type mongoDBInfraReconciler struct {
+	targetContext
+}
+
+// getMongoDBWatchedObjects provide list of object that needs to be watched to maintain MongoDB kogitoInfra resource
+func getMongoDBWatchedObjects() []framework.WatchedObjects {
+	return []framework.WatchedObjects{
+		{
+			GroupVersion: mongodb.SchemeGroupVersion,
+			AddToScheme:  mongodb.SchemeBuilder.AddToScheme,
+			Objects:      []runtime.Object{&mongodb.MongoDB{}},
+		},
+		{
+			Objects: []runtime.Object{&corev1.Secret{}},
+		},
+	}
+}
+
+// Reconcile reconcile Kogito infra object
+func (i *mongoDBInfraReconciler) Reconcile() (requeue bool, resultErr error) {
+	var mongoDBInstance *mongodb.MongoDB
+
+	if !infrastructure.IsMongoDBAvailable(i.client) {
+		return false, errorForResourceAPINotFound(&i.instance.Spec.Resource)
+	}
+
+	// Step 1: check whether user has provided custom mongoDB instance reference
+	mongoDBNamespace := i.instance.Spec.Resource.Namespace
+	mongoDBName := i.instance.Spec.Resource.Name
+	if len(mongoDBNamespace) == 0 {
+		mongoDBNamespace = i.instance.Namespace
+		log.Debugf("Namespace is not provided for infrastructure MongoDB resource %s, taking instance namespace(%s) as default", i.instance.Name, mongoDBNamespace)
+	}
+	if len(mongoDBName) == 0 {
+		return false, errorForResourceConfigError(i.instance, "No resource name given")
+	}
+
+	if mongoDBInstance, resultErr = i.loadDeployedMongoDBInstance(mongoDBName, mongoDBNamespace); resultErr != nil {
+		return false, resultErr
+	} else if mongoDBInstance == nil {
+		return false, errorForResourceNotFound("MongoDB", i.instance.Spec.Resource.Name, mongoDBNamespace)
+	}
+
+	log.Debugf("Got MongoDB instance %v", mongoDBInstance)
+	if mongoDBInstance.Status.Phase != mongodb.Running {
+		return false, errorForResourceNotReadyError(fmt.Errorf("mongoDB instance %s not ready. Waiting for Status.Phase == Running", mongoDBInstance.Name))
+	}
+	log.Info("MongoDB instance is running")
+	if resultErr = i.updateMongoDBAppPropsInStatus(mongoDBInstance); resultErr != nil {
+		return true, resultErr
+	}
+	if resultErr = i.updateMongoDBEnvVarsInStatus(mongoDBInstance); resultErr != nil {
+		return true, resultErr
```

Comment:
This error will immediately reconcile?

Reply:
why would it ? 
there is that https://github.com/kiegroup/kogito-cloud-operator/blob/f7ff09496d376272a5fb4a1500b4a783f2e47b20/pkg/controller/kogitoinfra/kogitoinfra_controller.go#L123

Reply:
Yup, we need to refactor this interface: https://issues.redhat.com/browse/KOGITO-3987